### PR TITLE
Angular routes to Django, Django routes to Angular (fixes #560)

### LINKED ISSUE #560: Angular/Django routing
Because we still have most of our views in Django we need to create infrastructure, that will allow us to at the same time use Angular views and Django views.

To achieve that we will create two separate roots (for Django and Angular) and everything, that will be outhside them will redirect to will redirect to each other.

For development environment it will be:
* Django `http://localhost:8000/o/`
* Django API `http://localhost:8000/api/`
* Angular `http://localhost:4200/`

For production environment it will be:
* Django `http://volontuloapp.org/o/`
* Django API `http://volontuloapp.org/api/`
* Django Angular `http://volontuloapp.org/` but except `o/` and `api/`

### PR DESCRIPTION
**Pull request by magul**
__Story / Bug id:__ https://github.com/CodeForPoznan/volontulo/issues/560

__Reference person:__ @magul

__Description:__

Django routes to Angular, Angular routes to Django

__New imports / dependencies:__

* N/A

__Unit Tests:__

* Multiple Django tests updated, Angular tests (`/frontend/src/app/redirect.component.spec.ts`)

__What tests do I need to run to validate this change:__

Build docker/vagrant environment and see, that whatever route You take on Angular it will be redirect to Django (with `/o` prefix).

### CODE REVIEW

File: frontend/src/app/redirect.component.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  template: ''
+})
+export class RedirectComponent {
+  constructor(
+    private router: Router,
+    @Inject(WindowToken) private window: any
+  ) {
+    const angularUrlSuffix = this.router.routerState.snapshot.url;
+    const djangoUrlSuffix = angularUrlSuffix === '/' ? '' : angularUrlSuffix;
+    this.window.location.href = `${environment.djangoRoot}${djangoUrlSuffix}`;
```
Comment (Bosmanfrx) [inline frontend/src/app/redirect.component.ts]: maybe we could use Location.go() here?

Comment (magul) [inline frontend/src/app/redirect.component.ts]: There' s no possibility to navigate to external url using location service (I checked And instead getting http://localhost:8000 I'm getting http://localhost:4200/http://localhost:8000

See also: https://stackoverflow.com/q/34338440

Comment (Bosmanfrx) [inline frontend/src/app/redirect.component.ts]: I'm just a bit worried about device agnosticism, but if that is the case I think your solution is good enough :)

Comment (magul) [inline frontend/src/app/redirect.component.ts]: If You thinnk so, feel free to merge it. :)

**Pull request merged**